service/mail-sender.js

```diff
@@ -0,0 +1,34 @@
+import Bluebird from 'bluebird';
+import config from './config';
+import nodemailer from 'nodemailer';
+import pug from 'pug';
+
+const transporter = nodemailer.createTransport({
+	host: config.mail.host,
+	port: config.mail.port,
+	secure: true,
+	auth: {
+		user: config.mail.username,
+		pass: config.mail.password
+	}
+});
+
+Bluebird.promisifyAll(transporter);
+
+const sender = (recipient, subject, view, params) => {
```

Comment:
Something to consider: Allowing pre-compiled Pug views. (E.g. allowing the `view` parameter to be a function.)